package.json

```diff
@@ -46,6 +46,7 @@
     "estraverse": "^4.2.0",
     "esutils": "^2.0.2",
     "file-entry-cache": "^2.0.0",
+    "functional-red-black-tree": "^1.0.1",
```

Comment:
I marked this PR as WIP because this module appears to be unmaintained. [Its readme](https://github.com/mikolalysenko/functional-red-black-tree) also mentions that the immutable approach ends up hurting performance for consumers that don't need immutability. We might be able to get an even larger performance boost if we implement our own balanced BST (or use a different maintained module, but I had trouble finding another balanced BST module that has the features we need).